core/market/decentralized/src/matcher/store.rs

```diff
@@ -218,4 +253,37 @@ impl SubscriptionStore {
             false => Err(DemandError::NotFound(demand_id.clone())),
         }
     }
+
+    /// Returns Offers SubscriptionId from vector, that don't exist in our database.
+    pub async fn filter_existing(
+        &self,
+        offers: Vec<SubscriptionId>,
+    ) -> Result<Vec<SubscriptionId>, QueryOfferError> {
+        Ok(futures::stream::iter(offers.into_iter())
+            .filter_map(|offer_id| {
+                let db = self.db.clone();
+                async move {
+                    match db
+                        .as_dao::<OfferDao>()
+                        .select(&offer_id, Utc::now().naive_utc())
+                        .await
+                    {
+                        Ok(offer_state) => match offer_state {
+                            OfferState::NotFound => Some(offer_id),
+                            _ => None,
+                        },
+                        Err(e) => {
+                            log::error!(
+                                "Can't get Offer [{}] from database. Error: {}",
+                                &offer_id,
+                                e
+                            );
+                            None
+                        }
+                    }
+                }
+            })
+            .collect::<Vec<SubscriptionId>>()
+            .await)
+    }
```

Comment:
Use batch_select. Make set difference.

Reply:
After rethinking: I can't use batch_select. It doesn't return unsubscribed and expired Offers. So if I remove from `offers` result of batch_select, it won't be what I expect here

Reply:
it was a pain for me that we do it that not optimal, so I've rewritten to use single select instead of multiple selects.
It was a broader effort of rewriting OfferDao